### PR TITLE
[docs] Hashicorp vault integration

### PR DESCRIPTION
**Pull request by aishwarya24**
@netlify /preview/integrations/hashicorp-vault/

### CODE REVIEW

**Review by ddhodge — APPROVED** [review]

File: docs/content/preview/integrations/hashicorp-vault.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+```go
+go build -o <build dir>/ysql-plugin cmd/ysql-plugin/main.go
+```
+
+Your YugabyteDB cluster should be up and running. Refer to [YugabyteDB Prerequisites](../../tools/#yugabytedb-prerequisites).
+
+## Run and configure the vault server
+
+The vault automatically registers the plugin if the directory of the binary of the plugin is provided as an input with the `dev-plugin-dir` flag. To run the vault server in development mode, use the `dev` flag. Set the `dev-root-token` flag which informs the vault to use the default vault token of root to login (this token is required in production mode).
```
Comment (HarshDaryani896) [inline docs/content/preview/integrations/hashicorp-vault.md:60]: Can we rephrase this to something like : 

Vault server can be run in development mode using `dev` flag, this makes it easy to experiment with Vault or start a Vault instance for development. Vault automatically registers the plugin if the directory of the binary of the plugin is provided as an input with the `dev-plugin-dir` flag. Set the `dev-root-token` flag which informs the vault to use the default vault token of root to login (this token is required in production mode).

Also, add the following warning as mentioned in [vault docs](https://developer.hashicorp.com/vault/docs/concepts/dev-server) : 

Warning: Never, ever, ever run a "dev" mode server in production. It is insecure and will lose data on every restart (since it stores data in-memory). It is only made for development or experimentation.

File: docs/content/preview/integrations/hashicorp-vault.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+For production mode, register the plugin as follows:
+
+```sh
+export SHA256=$(sha256sum <build dir>/ysql-plugin  | cut -d' ' -f1)
+
+vault write sys/plugins/catalog/database/ysql-plugin \
+    sha256=$SHA256 \
+    command="ysql-plugin"
+```
```
Comment (HarshDaryani896) [inline docs/content/preview/integrations/hashicorp-vault.md:82]: Put the following under `Setup` : 

For production mode, register the plugin as follows:

```sh
export SHA256=$(sha256sum <build dir>/ysql-plugin  | cut -d' ' -f1)

vault write sys/plugins/catalog/database/ysql-plugin \
    sha256=$SHA256 \
    command="ysql-plugin"
```

**Review by HarshDaryani896 — APPROVED** [review]
LGTM. Just some minor comments.

File: docs/content/preview/integrations/hashicorp-vault.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+- Enter the credentials:
+
+    ```sh
+    vault write database/config/yugabytedb plugin_name=ysql-plugin  \
+    host="127.0.0.1" \
+    port=5433 \
+    username="yugabyte" \
+    password="yugabyte" \
+    dbname="yugabyte" \
```
Comment (HarshDaryani896) [inline docs/content/preview/integrations/hashicorp-vault.md:94]: Change `dbname="yugabyte" \` to `db="yugabyte" \`
